frontend/src/utils/parseMessage.js

```diff
@@ -28,45 +28,60 @@ export default function parseMessage(message) {
 
   // Result node_bindings should be an array of qg_ids and kg_ids
   if ('results' in message) {
-    if (!Array.isArray(message.results[0].node_bindings)) {
-      message.answers = message.results;
-      delete message.results;
+    if (!Array.isArray(message.results)) {
+      throw Error('Message results should be an array.');
+    }
+    // results can be empty
+    if (message.results.length) {
+      if ('node_bindings' in message.results[0]) {
+        if (!Array.isArray(message.results[0].node_bindings)) {
+          // if results isn't in the newer format, we're gonna try and convert it
+          message.answers = message.results;
+          delete message.results;
+        }
+      } else {
+        throw Error('Error parsing message results. Results should have node_bindings and edge_bindings.');
```

Comment:
This error mentions edge_bindings but doesn't actually check for an edge_binding property.